stm32-modules/heater-shaker/tests/test_gcodes.cpp

```diff
@@ -687,3 +690,293 @@ SCENARIO("GetTemperatureDebug parser works") {
+            }
+        }
+    }
+
+    GIVEN("a string with p and i ok and d prefix only") {
+        std::string to_parse = "M301 P22.1 I55.1 D\r\n";
+        WHEN("calling parse") {
+            auto result = gcode::SetHeaterPIDConstants::parse(to_parse.cbegin(),
+                                                              to_parse.cend());
+
+            THEN("nothing should be parsed") {
+                REQUIRE(!result.first.has_value());
+                REQUIRE(result.second == to_parse.cbegin());
+            }
+        }
+    }
+
+    GIVEN("a string with p and i ok and d bad data") {
+        std::string to_parse = "M301 P22.1 I55.1 Dasdas\r\n";
+        WHEN("calling parse") {
+            auto result = gcode::SetHeaterPIDConstants::parse(to_parse.cbegin(),
+                                                              to_parse.cend());
+
+            THEN("nothing should be parsed") {
+                REQUIRE(!result.first.has_value());
+                REQUIRE(result.second == to_parse.cbegin());
+            }
+        }
+    }
+
+    GIVEN("a correct command") {
+        std::string to_parse = "M301 P22.1 I0.15 D-1.2\r\n";
+        WHEN("calling parse") {
+            auto result = gcode::SetHeaterPIDConstants::parse(to_parse.cbegin(),
+                                                              to_parse.cend());
+
+            THEN("a value should be parsed") {
+                REQUIRE(result.first.has_value());
+                REQUIRE_THAT(result.first.value().kp,
+                             Catch::Matchers::WithinAbs(22.1, .01));
+                REQUIRE_THAT(result.first.value().ki,
+                             Catch::Matchers::WithinAbs(0.15, 0.001));
+                REQUIRE_THAT(result.first.value().kd,
+                             Catch::Matchers::WithinAbs(-1.2, .01));
+                REQUIRE(result.second == (to_parse.cend() - 2));
+            }
+        }
+    }
+
+    GIVEN("a response buffer large enough for the response") {
+        std::string buffer(64, 'c');
+        WHEN("filling response") {
+            auto written = gcode::SetHeaterPIDConstants::write_response_into(
+                buffer.begin(), buffer.end());
+            THEN("the response should be written") {
+                REQUIRE_THAT(buffer, Catch::Matchers::StartsWith("M301 OK\n"));
+                REQUIRE(written == buffer.begin() + 8);
+            }
+        }
+    }
+
+    GIVEN("a response buffer not large enough for the response") {
+        std::string buffer(10, 'c');
+        WHEN("filling response") {
+            auto written = gcode::SetHeaterPIDConstants::write_response_into(
+                buffer.begin(), buffer.begin() + 5);
+            THEN("the response should be written only up to the size") {
+                REQUIRE_THAT(buffer, Catch::Matchers::Equals("M301 ccccc"));
+                REQUIRE(written == buffer.begin() + 5);
+            }
+        }
+    }
+}
+
+SCENARIO("SetHeaterPowerTest parser works") {
```

Comment:
Organizationally (I don't think that's a word..), it would be helpful for me to include the GCode in the scenario title.

Reply:
Oh, that's a good idea. I'll do that in the pr where i split this file  up.